web/src/engine/websites/CiaoPlus.ts

```diff
@@ -0,0 +1,213 @@
+                    })
+                );
+            };
+            function Is(e: number[], seed: number) : number[] {
+                const t = Ls(seed);
+                return e.map(o => [t.next().value, o]).sort((o, r) => + (o[0] > r[0]) - + (r[0] > o[0])).map(o => o[1]);
+            }
+
+            const Ls = function* (seed: number) {
+                const i = Uint32Array.of(seed);
+                for (; ;) i[0] ^= i[0] << 13,
+                i[0] ^= i[0] >>> 17,
+                i[0] ^= i[0] << 5,
+                yield i[0];
+            };
+
+            ctx.drawImage(image, 0, 0);
+            const o = getPieceDimension(image.width, image.height, COL_NUM);
+            ctx.clearRect(0, 0, o.width * COL_NUM, o.height * COL_NUM);
+            for (const piece of xs(COL_NUM, scramble_seed ?? 1)) {
+                ctx.drawImage(
+                    image,
+                    piece.source.x * o.width,
+                    piece.source.y * o.height,
+                    o.width,
+                    o.height,
+                    piece.dest.x * o.width,
+                    piece.dest.y * o.height,
+                    o.width,
+                    o.height
+                );
+            }
+
+        });
+    }
+
+    private async CreateRequest(endpoint: URL | string, postData: string[][] = undefined): Promise<Request> {
+        const url = typeof endpoint === 'string' ? new URL(endpoint) : endpoint;
+
+        //CALCULATE THE HASH for the request body OR request parameters
+        const params = postData ? Object.fromEntries(postData) : Object.fromEntries(url.searchParams);
+        const keys = Object.keys(params).sort();
+        const hashtable: string[] = [];
+        for (const key of keys) {
+            const keyHash = await this.SHA(key, 'SHA-256');
+            const valueHash = await this.SHA(params[key], 'SHA-512');
+            hashtable.push([keyHash, valueHash].join('_'));
+        }
+
+        //platform 3 = d294fcce0cc88587843099d85dd805aeef1b09a63b0db1dd3e4dc62a343c1db5_3bafbf08882a2d10133093a1b8433f50563b93c14acd05b79028eb1d12799027241450980651994501423a66c276ae26c43b739bc65c4e16b10c3af6c202aebb
+        //title_id_list 206 = ab64d73d61fd0aca64d927401adb3a267105307ee8de12dc2313443a2ce2cb5c_33d7bcca06466d44b59bcb0683b2542acf08e476e2a7faffb3800f6a01b0c13a9049999a42f3dd159db0d85c7208cfc67f933c927cf48878058f0931e97ee1a3
+        //version 6.0.0 = "5ca4f3850ccc331aaf8a257d6086e526a3b42a63e18cb11d020847985b31d188_6a2f355dc194a53a7d87513d236514b5e85bd3da886e37f93dcbda18f9d4204aa74b5c1eb2063547e6afa206d8bafe860896a271baa83a5e0a836fe44da2a7c3"
+
+        let bambihash = await this.SHA(hashtable.toString(), 'SHA-256'); //2cc9285228b63b58c6cc404082904170a7e4f89fb9c3cee00c53a7f67ad5ed6b
+        bambihash = await this.SHA(bambihash, 'SHA-512'); //ad63b1c89a6aa7bf0664adcc99d2fc212c77795f8c83b05c0b1ade00da16a1b375516d963de2e70aee53843ed8d0eeaf4c6ecc32732c3b7d9bda63f6d6a5916e
+
+        return postData ?
```

Comment:
Consider to provide different wrapper methods for GET and POST to reduce `if...else` complexity.